docs/architecture/adr-004-tips-contract.md

```diff
@@ -54,26 +55,31 @@ tip and paid to the contract.
 ```rust
 pub enum ServiceFee {
   Fixed {amount: Vec<Coin>},
-  Percentage{value: u64}
+  Percentage{value: u64, decimals: u32}
 }
 ```
 
-* The value of the percentage is represented by a `u64` cause float can't be used inside contracts. The Service fee will then be calculated as follow so:
-  * Service Fee = `tip_amount * value/100`
+The `Percentage` fee is represented by a:
+  * `value` field that identifies the number of percentage the admin applies (e.g. value = 2 means that the percentage can be 2%, 0,2%, etc..based on the 2nd field);
+  * `decimals` field represents the number of decimal places that come before the `value`.
+
+The `Percentage` fee will then be calculated as follows so: `tip_amount * (value*10^decimals)/(100*10^decimals)`
```

Comment:
Shouldn't it be `tip_amount * value /  10^decimals`?
value = 2, decimals = 1, then it is 0.2
value = 2, decimals = 2, then it is 0.02

Reply:
@dadamu You are right. Thanks for the correction.